### PR TITLE
Updating Event Class 

### PR DESCRIPTION
**Pull request by harshpalan**
Updating Event Class to accommodate changes from PR [61](https://github.com/soundata/soundata/pull/61) for Spatial Events. Now Event Class contains all the required attributes for Spatial Events.

### CODE REVIEW

File: soundata/annotations.py [hunk 1]
```diff
@@ -65,12 +99,33 @@ def __init__(
         validate_confidence(confidence)
         validate_unit(labels_unit, LABEL_UNITS)
         validate_unit(intervals_unit, TIME_UNITS)
+        validate_array_like(azimuth, np.ndarray, float, none_allowed=True)
+        validate_array_like(elevation, np.ndarray, float, none_allowed=True)
+        validate_array_like(distance, np.ndarray, float, none_allowed=True)
+        validate_array_like(cartesian_coord, np.ndarray, float, none_allowed=True)
+        validate_lengths_equal([azimuth, elevation, distance, cartesian_coord])
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: Let's add intervals here to impose coherence

File: soundata/annotations.py [hunk 2]
```diff
@@ -65,12 +99,33 @@ def __init__(
...
+        validate_azimuth(azimuth, azimuth_unit, allow_none=True)
+        validate_distance(distance, allow_none=True)
+        validate_azimuth(elevation, elevation_unit, allow_none=True)
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: Is this right? Is because the limits are the same?

File: soundata/annotations.py [hunk 3]
```diff
@@ -272,3 +327,45 @@ def validate_unit(unit, unit_values, allow_none=False):
...
+
+def validate_cartesian_coord(cartesian_coord, allow_none=False):
+    if allow_none:
+        return
+
+    if not np.shape(cartesian_coord)[1] == 3:
+        raise ValueError(
+            f"cartesian coordinates should have three columns corresponding to x, y and z coordinates. "
+            f"Found {len(cartesian_coord)} columns instead."
+        )
```
Comment (magdalenafuentes) [inline soundata/annotations.py:399]: Let's check for distances>0 inside coordinates as well

File: soundata/annotations.py [hunk 4]
```diff
@@ -50,12 +63,33 @@ class Events(Annotation):
         confidence (np.ndarray or None): array of confidence values, float in [0, 1]
         labels_unit (str): labels unit, one of LABELS_UNITS
         intervals_unit (str): intervals unit, one of TIME_UNITS
+        azimuth (np.ndarray or None):
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: Missing description on azimuth, elevation and distance

File: soundata/annotations.py [hunk 5]
```diff
@@ -272,3 +329,45 @@ def validate_unit(unit, unit_values, allow_none=False):
...
+            "distance should be bigger or equal than zero. " + "Found negative values."
+        )
+
+
+def validate_cartesian_coord(cartesian_coord, allow_none=False):
+    if allow_none and cartesian_coord is None:
+        return
+
+    if not np.shape(cartesian_coord)[1] == 3:
+        raise ValueError(
```
Comment (magdalenafuentes) [inline soundata/annotations.py:396]: We're not testing this line

**Review by magdalenafuentes — CHANGES_REQUESTED** [review]
Hey @harshpalan this is looking great, just a few more things and we merge:
- Azimuth, elevation and distance descriptions in function readme are missing
- There's line 370 not being tested
- See my comments below about checking distances in coordinates

File: soundata/annotations.py (hunk 2 above)
Comment (harshpalan) [reply soundata/annotations.py]: Added a new function validation function for elevation.

File: soundata/annotations.py [hunk 6]
```diff
@@ -272,3 +333,67 @@ def validate_unit(unit, unit_values, allow_none=False):
...
+                + "Found values outside that interval."
+            )
+
+
+def validate_elevation(elevation, elevation_unit=None, allow_none=False):
+    if allow_none and not elevation_unit:
+        return
+
+    if elevation_unit == "radians":
+        if any([a < -2 * np.pi for a in elevation]) or any(
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: Should be `pi/2`

**Review by magdalenafuentes — APPROVED** [review]

**Pull request merged**